plugins/cpp_lsp/service/src/cpplspservice.cpp

```diff
@@ -0,0 +1,658 @@
+    [](std::pair<std::string, std::int32_t> item)
+    {
+      return item.first;
+    }
+  );
+
+  return diagramTypes;
+}
+
+void CppLspServiceHandler::getDiagram(
+  pt::ptree& responseTree_,
+  const pt::ptree& params_)
+{
+  DiagramParams diagramParams;
+  diagramParams.readNode(params_);
+
+  Diagram diagramResult;
+  if (!diagramParams.position)
+  {
+    diagramResult = fileDiagram(diagramParams);
+  }
+  else
+  {
+    diagramResult = nodeDiagram(diagramParams);
+  }
+
+  responseTree_.put("result", diagramResult);
+}
+
+void CppLspServiceHandler::getModuleDiagram(
+  pt::ptree& responseTree_,
+  const pt::ptree& params_)
+{
+  getDiagram(responseTree_, params_);
+}
+
+Diagram CppLspServiceHandler::fileDiagram(
+  const DiagramParams& params_)
+{
+  model::FilePtr file = _transaction([&, this](){
+    return _db->query_one<model::File>(
+      odb::query<model::File>::path == params_.textDocument.uri);
+  });
+
+  if (!file)
+    return std::string();
+
+  const static std::map<std::string, std::int32_t> diagramTypes = {
+    {"Internal architecture of this module",
+      language::CppServiceHandler::SUBSYSTEM_DEPENDENCY},
+    {"This module depends on",
+      language::CppServiceHandler::EXTERNAL_DEPENDENCY},
+    {"Users of this module", language::CppServiceHandler::EXTERNAL_USERS},
+    {"Include Dependency", language::CppServiceHandler::INCLUDE_DEPENDENCY},
+    {"Component Users", language::CppServiceHandler::COMPONENT_USERS},
+    {"Interface Diagram", language::CppServiceHandler::INTERFACE},
+  };
+
+  auto diagramTypeIt = diagramTypes.find(params_.diagramType);
+  if(diagramTypeIt == diagramTypes.end())
+    return std::string();
+
+  auto graph = _cppService.returnFileDiagram(
+    std::to_string(file->id),
+    diagramTypeIt->second);
+
+  if (graph.nodeCount() != 0)
+  {
+    addPathToIdInFileDiagram(graph, std::to_string(file->id));
+    return graph.output(util::Graph::SVG);
+  }
+  return std::string();
+}
+
+Diagram CppLspServiceHandler::nodeDiagram(
+  const DiagramParams& params_)
+{
+  language::AstNodeInfo astNodeInfo;
+  core::FilePosition cppPosition;
+
+  model::FilePtr file = _transaction([&, this](){
+    return _db->query_one<model::File>(
+      odb::query<model::File>::path == params_.textDocument.uri);
+  });
+
+  if (!file)
+    return std::string();
+
+  cppPosition.file = std::to_string(file->id);
+  cppPosition.pos.line = params_.position->line;
+  cppPosition.pos.column = params_.position->character;
+
+  _cppService.getAstNodeInfoByPosition(astNodeInfo, cppPosition);
+
+  const static std::map<std::string, std::int32_t> diagramTypes = {
+    {"Function call diagram", language::CppServiceHandler::FUNCTION_CALL},
+    {"Detailed class diagram", language::CppServiceHandler::DETAILED_CLASS},
+    {"Class collaboration diagram",
+      language::CppServiceHandler::CLASS_COLLABORATION},
+  };
+
+  auto diagramTypeIt = diagramTypes.find(params_.diagramType);
+  if(diagramTypeIt == diagramTypes.end())
```

Comment:
Please follow the coding convention regarding the brackets everywhere.

Reply:
I am not completely sure which part you mean.
Was it the initializer list for the `diagramTypes`?

Reply:
I guess @intjftw  just meant that a space is missing after the `if` keyword.